### PR TITLE
Fix etcd and http get bugs

### PR DESCRIPTION
**Pull request by HunDunDM**
* check http code
* update etcd client config

### CODE REVIEW

File: pkg/keyvisual/decorator/utils.go [hunk 1]
```diff
@@ -28,14 +29,25 @@ const (
 	etcdGetTimeout = time.Second
 )
 
+var (
+	ErrNS          = errorx.NewNamespace("error.keyvisual")
+	ErrNSDecorator = ErrNS.NewSubNamespace("decorator")
+	ErrNotOkStatus = ErrNSDecorator.NewType("not_ok_status")
```
Comment (breeswish) [inline pkg/keyvisual/decorator/utils.go]: How about calling it ErrTiDBHTTPRequestFailed?

File: pkg/keyvisual/input/api.go [hunk 2]
```diff
@@ -18,12 +18,21 @@ import (
...
 
+	"github.com/joomcode/errorx"
+
 	regionpkg "github.com/pingcap-incubator/tidb-dashboard/pkg/keyvisual/region"
 )
 
+var (
+	ErrNS          = errorx.NewNamespace("error.keyvisual")
+	ErrNSInput     = ErrNS.NewSubNamespace("input")
+	ErrNotOkStatus = ErrNSInput.NewType("not_ok_status")
```
Comment (breeswish) [inline pkg/keyvisual/input/api.go]: How about calling it ErrPDHTTPRequestFailed?

File: pkg/keyvisual/input/api.go [hunk 3]
```diff
@@ -118,9 +127,16 @@ func NewAPIPeriodicGetter(pdAddr string) regionpkg.RegionsInfoGenerator {
...
+		} else if resp.StatusCode != http.StatusOK {
+			defer resp.Body.Close()
+			body, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return nil, err
+			}
+			return nil, ErrPDHTTPRequestFailed.New("http status code: %d, msg: %s", resp.StatusCode, string(body))
 		}
-		return nil, err
+		return read(resp.Body)
```
Comment (rleungx) [inline pkg/keyvisual/input/api.go]: still need `resp.Body.Close()`?

Comment (HunDunDM) [reply pkg/keyvisual/input/api.go]: read is a generic function that has handled Close.
```
func read(stream io.ReadCloser) (*RegionsInfo, error) {
	defer stream.Close()
```

**Pull request closed without merging**